vaas/vaas/cluster/api.py

```diff
@@ -148,3 +153,92 @@ class Meta:
+    def __init__(self, pk=None, varnish_ids=None, status=None, output=None):
+        self.pk = pk
+        self.id = pk
+        self.varnish_ids = varnish_ids
+        self.status = status
+        self.output = output
+
+    def __repr__(self):
+        return '{}'.format(self.__dict__)
+
+
+class CommandInputValidation(Validation):
+    def is_valid(self, bundle, request=None):
+        errors = {}
+        try:
+            varnishes = [int(varnish_id) for varnish_id in set(bundle.data.get('varnish_ids', []))]
+            assert VarnishServer.objects.filter(pk__in=varnishes).count() == len(varnishes)
+        except AssertionError:
+            errors['varnish_ids'] = 'Some of provided varnish identifiers does not exists'
+        except ValueError:
+            errors['varnish_ids'] = 'Unexpected type, identifiers have got to be an integer'
+        except:  # noqa
+            errors['varnish_ids'] = 'Unexpected value'
+        else:
+            task = AsyncResult(bundle.data['pk'])
+            if task.args and set(task.args[0]) != set(varnishes):
+                errors['__all__'] = 'Command: %s has been already ordered with another varnish_ids set' \
+                                    % bundle.data['pk']
+
+        return errors
+
+
+class ConnectCommandResource(Resource):
+    pk = fields.CharField(attribute='id', readonly=True)
+    varnish_ids = fields.ListField(attribute='varnish_ids')
+    status = fields.CharField(attribute='status', readonly=True, blank=True, null=True)
+    output = fields.DictField(attribute='output', readonly=True, blank=True, null=True)
+
+    class Meta:
+        resource_name = 'connect-command'
+        list_allowed_methods = []
+        detail_allowed_methods = ['put', 'get']
+        authorization = DjangoAuthorization()
+        authentication = VaasMultiAuthentication(ApiKeyAuthentication())
+        validation = CommandInputValidation()
+        fields = ['input', 'status', 'output']
+        include_resource_uri = False
+        always_return_data = True
+
+    def obj_update(self, bundle, **kwargs):
+        bundle.data['pk'] = kwargs['pk']
+        self.run_validation(bundle)
+        raise NotFound()
+
+    def obj_create(self, bundle, **kwargs):
+        bundle.obj = CommandResource(pk=kwargs['pk'])
+        bundle = self.full_hydrate(bundle)
+        task = connect_command.apply_async([bundle.obj.varnish_ids], task_id=bundle.obj.pk)
+        bundle.obj.status = task.status
+        bundle.obj.output = task.result
+        return bundle
+
+    def hydrate_varnish_ids(self, bundle):
+        bundle.obj.varnish_ids = bundle.data['varnish_ids']
+        return bundle
+
+    def obj_get(self, bundle, **kwargs):
+        task = AsyncResult(kwargs['pk'])
+        varnish_ids = []
+        if task.args and len(task.args):
+            varnish_ids = task.args[0]
+        return CommandResource(kwargs['pk'], varnish_ids, task.status, output=task.result)
+
+    def get_object_list(self, request):
```

Comment:
It's a part of a contract. If you define your own resource that is not based on modelresource, you have to implement some methods. I will verify if it can be safely removed.

Reply:
Thanks for checking. I also curious  it is necessary.

Reply:
I've checked - and removed this method and overriden the URL for listing purposes.